src/interface/swarm.hpp

```diff
@@ -285,18 +295,43 @@ inline SwarmVariablePack<T> Swarm::PackVariables(const std::vector<std::string>
 }
 
 template <class T>
-inline SwarmVariablePack<T> Swarm::PackAllVariables(PackIndexMap &vmap) {
+inline SwarmVariablePack<T>
+Swarm::PackAllVariables(PackIndexMap &vmap, ParArrayND<int> &pack_indices_shapes) {
```

Comment:
As mentioned above, some doc on `pack_indices_shapes` would be helpful potentially even as function docstring

Reply:
I added the inline description of `pack_indices_shapes` to this function, let me know if I could do a better job describing it